runtime/cudaq/cudaq.cpp

```diff
@@ -36,98 +36,91 @@ cudaq::MPIPlugin *getMpiPlugin(bool unsafe) {
-                    activatedInterfaceLibFile.c_str());
-        g_plugin = std::make_unique<cudaq::MPIPlugin>(
-            activatedInterfaceLibFile.c_str());
-      } else {
-        const auto pluginsPath = cudaqLibPath.parent_path() / "plugins";
+      const auto pluginsPath = cudaqLibPath.parent_path() / "plugins";
 #if defined(__APPLE__) && defined(__MACH__)
-        const std::string libSuffix = "dylib";
+      const std::string libSuffix = "dylib";
 #else
-        const std::string libSuffix = "so";
+      const std::string libSuffix = "so";
 #endif
-        // The builtin (native) plugin if present
-        const auto pluginLibFile =
-            pluginsPath / fmt::format("libcudaq-comm-plugin.{}", libSuffix);
-        if (std::filesystem::exists(pluginLibFile) &&
-            cudaq::MPIPlugin::isValidInterfaceLib(pluginLibFile.c_str())) {
-          cudaq::info("Load builtin MPI comm plugin at '{}'",
-                      pluginLibFile.c_str());
-          g_plugin = std::make_unique<cudaq::MPIPlugin>(pluginLibFile.c_str());
-        }
+      // The builtin (native) plugin if present
+      const auto pluginLibFile =
+          pluginsPath / fmt::format("libcudaq-comm-plugin.{}", libSuffix);
+      if (std::filesystem::exists(pluginLibFile) &&
+          cudaq::MPIPlugin::isValidInterfaceLib(pluginLibFile.c_str())) {
+        cudaq::info("Load builtin MPI comm plugin at '{}'",
+                    pluginLibFile.c_str());
+        g_plugin = std::make_unique<cudaq::MPIPlugin>(pluginLibFile.c_str());
       }
     }
   }
   if (!g_plugin) {
     if (unsafe)
       return nullptr;
-    else
-      throw std::runtime_error(
-          "No MPI support can be found when attempted to use cudaq::mpi APIs. "
-          "Please refer to the documentation for instructions to activate MPI "
-          "support.");
+    throw std::runtime_error(
+        "No MPI support can be found when attempted to use cudaq::mpi APIs. "
+        "Please refer to the documentation for instructions to activate MPI "
+        "support.");
   }
 
   return g_plugin.get();
 };
 
-bool available() {
-  auto *commPlugin = getMpiPlugin(/*unsafe=*/true);
-  return commPlugin != nullptr;
-}
+bool available() { return getMpiPlugin(/*unsafe=*/true) != nullptr; }
 
 void initialize() {
-  auto *commPlugin = getMpiPlugin();
-  commPlugin->initialize();
+  if (auto *commPlugin = getMpiPlugin()) {
+    commPlugin->initialize();
+  }
 }
 
 void initialize(int argc, char **argv) {
-  auto *commPlugin = getMpiPlugin();
-  commPlugin->initialize(argc, argv);
-  const auto pid = commPlugin->rank();
-  const auto np = commPlugin->num_ranks();
-  if (pid == 0)
-    cudaq::info("MPI Initialized, nRanks = {}", np);
+  if (auto *commPlugin = getMpiPlugin()) {
+    commPlugin->initialize(argc, argv);
+    const auto pid = commPlugin->rank();
+    const auto np = commPlugin->num_ranks();
+    if (pid == 0)
+      cudaq::info("MPI Initialized, nRanks = {}", np);
+  }
```

Comment:
I'm not sure I follow the need for the change here. `getMpiPlugin()` is guaranteed to return a valid pointer (unless we're setting `unsafe` to true, and we're not doing that), so the added check seems unnecessary IMO.